addons/io/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/HueEmulationServlet.java

```diff
@@ -0,0 +1,521 @@
+            out.close();
+        } catch (ItemNotFoundException e) {
+            logger.warn("Item not found: " + id);
+            apiServerError(req, resp, HueErrorResponse.NOT_AVAILABLE, "Item not found " + id);
+        }
+    }
+
+    /**
+     * Hue API call to get a listing of all lights
+     *
+     * @param req
+     * @param resp
+     * @throws IOException
+     */
+    public void apiLights(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+        PrintWriter out = resp.getWriter();
+        Gson gson = new Gson();
+        out.write(gson.toJson(getHueDeviceNames()));
+        out.close();
+    }
+
+    /**
+     * HUE API call to get the Data Store of the bridge (only lights supported for now)
+     *
+     * @param req
+     * @param resp
+     * @throws IOException
+     */
+    public void apiDataStore(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+        PrintWriter out = resp.getWriter();
+        Gson gson = new Gson();
+        HueDataStore ds = new HueDataStore();
+        ds.lights = getHueDevices();
+        out.write(gson.toJson(ds));
+    }
+
+    /**
+     * Hue API call to configure a user
+     *
+     * @param req
+     * @param resp
+     * @throws IOException
+     */
+    public void apiConfig(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+        if (!req.getMethod().equals(METHOD_POST)) {
+            apiServerError(req, resp, HueErrorResponse.METHOD_NOT_AVAILABLE, "Only POST allowed for this resource");
+            return;
+        }
+        PrintWriter out = resp.getWriter();
+        Gson gson = new Gson();
+
+        HueCreateUser user = gson.fromJson(req.getReader(), HueCreateUser.class);
+        logger.debug("Create user: " + user.devicetype);
+        if (user.username == null || user.username.length() == 0) {
+            user.username = UUID.randomUUID().toString();
+        }
+        addUser(user.username);
+        String response = String.format(NEW_CLIENT_RESP, user.username);
+        out.write(response);
+        out.close();
+    }
+
+    /**
+     * Hue API error response
+     *
+     * @param req
+     * @param resp
+     * @param error
+     * @param description
+     * @throws IOException
+     */
+    public void apiServerError(HttpServletRequest req, HttpServletResponse resp, int error, String description)
+            throws IOException {
+        PrintWriter out = resp.getWriter();
+        Gson gson = new Gson();
+        HueErrorResponse e = new HueErrorResponse(error, req.getRequestURI(), description);
+        out.write(gson.toJson(e));
+    }
+
+    /**
+     * Generates the XML Discovery document
+     *
+     * @return
+     *         XML document
+     */
+    public void apiDiscoveryXML(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+        String discoString = IOUtils.toString(getClass().getClassLoader().getResourceAsStream("discovery.xml"),
```

Comment:
Do this as field in the class, then the file is only read once and then read from memory.